### PR TITLE
Added tunnel capability struct to json, json to struct and free api

### PR DESCRIPTION
**Pull request by utkarsha-elear**

### CODE REVIEW

File: coco_std_data_tunnel_types.h [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+}coco_std_tunnel_close_t;
+
+typedef enum {
+  COCO_STD_CMD_TUNNEL_MIN = -1,
+  COCO_STD_CMD_TUNNEL_OPEN,
+  COCO_STD_CMD_TUNNEL_CLOSE,
+  COCO_STD_CMD_TUNNEL_MAX
+} coco_std_cmd_tunnel_t;
+
+typedef enum{
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.h]: Move this enum to `coco_std_api.h`

File: coco_std_data_tunnel_types.h [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ *            structure for a given command ID.
+ * @param     commandId: The command the structure belongs to
+ * @param     inStruct : Pointer to Input Structure to be deallocated
+ * @return    0 on Success,
+ *            -1 on Failure
+ ***********************************************/
+int coco_internal_tunnel_free_handler(int commandId, void *inStruct);
+
+#endif // COCO_STD_DATA_TUNNEL_TYPES_H_INCLUDED
+
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.h]: Remove the extra blank line from the end of the file. Have just one blank line

File: coco_std_data_tunnel_types.h [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  COCO_STD_TUNNEL_PROTOCOL_MIN = -1,
+  COCO_STD_TUNNEL_PROTOCOL_TCP,
+  COCO_STD_TUNNEL_PROTOCOL_UDP,
+  COCO_STD_TUNNEL_PROTOCOL_MAX,
+  COCO_STD_TUNNEL_PROTOCOL_UBOUND = 0x7FFF
+}coco_std_tunnel_protocol_type_t;
+
+typedef enum {
+  COCO_STD_ATTR_TUNNEL_MIN = -1,
+  COCO_STD_ATTR_TUNNEL_IPV4_FWD_HOST,
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.h]: Remove `IPV4` from the names

File: coco_std_data_tunnel_types.h [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  COCO_STD_CMD_TUNNEL_MAX
+} coco_std_cmd_tunnel_t;
+
+typedef enum{
+  COCO_STD_TUNNEL_PROTOCOL_MIN = -1,
+  COCO_STD_TUNNEL_PROTOCOL_TCP,
+  COCO_STD_TUNNEL_PROTOCOL_UDP,
+  COCO_STD_TUNNEL_PROTOCOL_MAX,
+  COCO_STD_TUNNEL_PROTOCOL_UBOUND = 0x7FFF
+}coco_std_tunnel_protocol_type_t;
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.h]: Rename to `coco_std_tunnel_type_t`

File: coco_std_data_tunnel_types.c [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+
+/*************************************************************************************
+ *                          GLOBAL VARIABLES                                         *
+ *************************************************************************************/
+
+/*************************************************************************************
+ *                          LOCAL VARIABLES                                          *
+ *************************************************************************************/
+static const coco_internal_cmd_json_to_struct_fn_t tunnelCmdJsonToStruct[] = {
+  [COCO_STD_CMD_TUNNEL_OPEN]  = set_tunnel_open_cmd_json_to_struct,
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.c]: Rename to `tunnel_open_cmd_json_to_struct` and `tunnel_close_cmd_json_to_struct`

File: coco_std_data_tunnel_types.c [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+  [COCO_STD_CMD_TUNNEL_CLOSE] = set_tunnel_close_cmd_json_to_struct
+};
+
+static coco_internal_cap_struct_to_json_hndlr_fn_t tunnelCmdStructToJson[] = {
+  [COCO_STD_CMD_TUNNEL_OPEN]  =  set_tunnel_open_struct_to_json,
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.c]: Rename to `tunnel_open_cmd_struct_to_json` and `tunnel_close_cmd_struct_to_json`

File: coco_std_data_tunnel_types.c [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+  [COCO_STD_CMD_TUNNEL_CLOSE] =  set_tunnel_close_struct_to_json
+};
+
+static coco_internal_free_cmd_hndlr_t tunnelCmdHandlerLUT[] = {
+  [COCO_STD_CMD_TUNNEL_OPEN]  = coco_internal_free_open_tunnel_cmd,
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.c]: Rename to `tunnel_open_cmd_free` and `tunnel_close_cmd_free`

File: coco_std_data_tunnel_types.c [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+  [COCO_STD_CMD_TUNNEL_OPEN]  = set_tunnel_open_cmd_json_to_struct,
+  [COCO_STD_CMD_TUNNEL_CLOSE] = set_tunnel_close_cmd_json_to_struct
+};
+
+static coco_internal_cap_struct_to_json_hndlr_fn_t tunnelCmdStructToJson[] = {
+  [COCO_STD_CMD_TUNNEL_OPEN]  =  set_tunnel_open_struct_to_json,
+  [COCO_STD_CMD_TUNNEL_CLOSE] =  set_tunnel_close_struct_to_json
+};
+
+static coco_internal_free_cmd_hndlr_t tunnelCmdHandlerLUT[] = {
```
Comment (varun-elear) [inline coco_std_data_tunnel_types.c]: Rename to `tunnelCmdFree`

File: coco_std_util.h [hunk 9]
```diff
@@ -61,6 +61,8 @@
 #define COCO_STD_CMD_JSON_NOTIFY_TYPE             "notificationType"
 #define COCO_STD_CMD_JSON_EVENT_TYPE              "eventType"
 
+#define COCO_STD_CMD_JSON_TUNNEL_CHANNEL_PORT     "channelPort"
```
Comment (varun-elear) [inline coco_std_util.h]: Add a comment - `Macros for field names in JSON for tunnel commands`

File: coco_std_data_tunnel_types.c [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+/*===================================================================================*/
+#include "elearcommon/allocate.h"
+#include "elearcommon/logger.h"
+#include "elearcommon/json_services.h"
+#include "elearcommon/util.h"
+#include "elearcommon/ec_global_vars.h"
+#include "coco_std_util.h"
+#include "coco_std_api.h"
+#include "coco_std_data_tunnel_types.h"
+#include "coco_std_data_resource_cmd_util.h"
```
[the remark this answers is not in the document]
Comment (utkarsha-elear) [reply coco_std_data_tunnel_types.c]: yes

[the remark this answers is not in the document]
Comment (utkarsha-elear) [reply coco_std_data_tunnel_types.c]: Yes

File: coco_std_api.h [hunk 11]
```diff
@@ -577,6 +577,15 @@ typedef enum {
   COCO_STD_ERRNO_MAX
 } coco_std_errno_t;
 
+// Tunnel enum
+typedef enum {
+  COCO_STD_TUNNEL_PROTOCOL_MIN = -1,
+  COCO_STD_TUNNEL_PROTOCOL_TCP,
+  COCO_STD_TUNNEL_PROTOCOL_UDP,
+  COCO_STD_TUNNEL_PROTOCOL_MAX,
+  COCO_STD_TUNNEL_PROTOCOL_UBOUND = 0x7FFF
```
Comment (varun-elear) [inline coco_std_api.h]: Should be 0x7F

**Pull request merged**